### PR TITLE
v3.0.0rc1 release

### PR DESCRIPTION
**Pull request by 2pd**
### Changed
- Redesign of Websocket part. Please consult `README.md` for details on its new usage.

### Added
- Add Websocket API

### CODE REVIEW

**Review by aisling-2 — APPROVED** [review]

File: binance/websocket/binance_socket_manager.py [hunk 1]
```diff
@@ -1,83 +1,100 @@
...
-        if not is_live:
-            payload_obj = json.loads(payload.decode("utf8"))
-
-            if is_combined:
-                factory_url = factory_url + "?streams=" + payload_obj["params"]
-            else:
-                factory_url = factory_url + "/" + payload_obj["params"]
-            payload = None
+    def create_ws_connection(self):
+        self.logger.debug("Start connect to WebSocket Server: %s", self.stream_url)
```
Comment (urbanogt) [inline binance/websocket/binance_socket_manager.py:37]: Start connect to WebSocket Server
->
Creating connection with WebSocket Server

File: binance/websocket/binance_socket_manager.py [hunk 2]
```diff
@@ -1,83 +1,100 @@
...
+        self.ws = create_connection(self.stream_url)
+        self.logger.debug("Created WebSocket connection to: %s", self.stream_url)
```
Comment (urbanogt) [inline binance/websocket/binance_socket_manager.py:39]: Created WebSocket connection to:
->
WebSocket connection has been established

File: binance/websocket/binance_socket_manager.py [hunk 3]
```diff
@@ -1,83 +1,100 @@
...
+    def read_data(self):
+        data = ""
+        while True:
+            try:
+                op_code, frame = self.ws.recv_data_frame(True)
+            except WebSocketException as e:
+                if isinstance(e, WebSocketConnectionClosedException):
+                    self.logger.error("lost websocket connection")
+                else:
+                    self.logger.error("websocket exception: %s", e)
```
Comment (urbanogt) [inline binance/websocket/binance_socket_manager.py:61]: A bit nitpick, but sometimes you use % formatting for strings, and in others .format . I would keep the same everywhere just for consistency with the style.

In the logs, sometimes you start with uppercase and others lowercase.

File: binance/websocket/spot/websocket_api/__init__.py [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+    from binance.websocket.spot.websocket_api.trade import cancel_open_orders
+    from binance.websocket.spot.websocket_api.trade import new_oco_order
+    from binance.websocket.spot.websocket_api.trade import get_oco_order
+    from binance.websocket.spot.websocket_api.trade import cancel_oco_order
+    from binance.websocket.spot.websocket_api.trade import get_open_oco_orders
+
+    # User Data Stream
+    from binance.websocket.spot.websocket_api.user_data import user_data_start
+    from binance.websocket.spot.websocket_api.user_data import user_data_ping
+    from binance.websocket.spot.websocket_api.user_data import user_data_stop
```
Comment (urbanogt) [inline binance/websocket/spot/websocket_api/__init__.py:70]: Interesting way to split a big class, although I would use different names in the files to let user know that he shouldn't import them directly.

    from binance.websocket.spot.websocket_api.market import ping_connectivity
->
    from binance.websocket.spot.websocket_api._market import ping_connectivity

**Review by urbanogt — APPROVED** [review]
LGTM

File: binance/websocket/spot/websocket_api/__init__.py (hunk 4 above)
Comment (2pd) [reply binance/websocket/spot/websocket_api/__init__.py:70]: that's very good point

**Review by urbanogt — APPROVED** [review]

**Pull request merged**